packages/flame/lib/src/effects/sequence_effect.dart

```diff
@@ -27,35 +46,36 @@ import 'package:flame/src/effects/effect.dart';
 /// effect depends on the timings of individual effects, and cannot be
 /// represented as a regular effect controller.
 class SequenceEffect extends Effect {
-  factory SequenceEffect(
+  SequenceEffect(
     List<Effect> effects, {
     bool alternate = false,
     bool infinite = false,
     int repeatCount = 1,
-    void Function()? onComplete,
-  }) {
-    assert(effects.isNotEmpty, 'The list of effects cannot be empty');
-    assert(
-      !(infinite && repeatCount != 1),
-      'Parameters infinite and repeatCount cannot be specified simultaneously',
-    );
-    EffectController ec = _SequenceEffectEffectController(
-      effects,
-      alternate: alternate,
-    );
-    if (infinite) {
-      ec = InfiniteEffectController(ec);
-    } else if (repeatCount > 1) {
-      ec = RepeatedEffectController(ec, repeatCount);
-    }
-    effects.forEach((e) => e.removeOnFinish = false);
-    return SequenceEffect._(ec, onComplete: onComplete)..addAll(effects);
-  }
-
-  SequenceEffect._(
-    super.ec, {
     super.onComplete,
-  });
+  })  : assert(effects.isNotEmpty, 'The list of effects cannot be empty'),
+        assert(
+          !(infinite && repeatCount != 1),
+          'Parameters infinite and repeatCount cannot be specified '
+          'simultaneously',
+        ),
+        _effects = effects,
+        super(
+          _createController(
+            effects: effects,
+            alternate: alternate,
+            infinite: infinite,
+            repeatCount: repeatCount,
+          ),
+        );
+
+  final List<Effect> _effects;
+
+  @override
+  void onMount() {
+    super.onMount();
+
+    addAll(_effects);
```

Comment:
Any reason these are added in `onMount` instead of in the constructor now? It probably would break if this effect was added, removed and added again?

Reply:
No particular reason other than my brain is still waking up 😅 

Changed it!